### PR TITLE
feat: improve client interface

### PR DESCRIPTION
**Pull request by samsja**
- [x] Allow `Client(host='http://localhost:1234')`
- [x] handle duplicate ex : `Client(host='http://localhost:1234',port = 4321)`
- [x] SKIP todo in another pr handle `tls` for `grpc` and `websocket`
- [x] rename `websocket` to `ws`
- [x] rename `https` to `tls`
- [x] add docs 

linked issue : 
https://github.com/jina-ai/jina/issues/4493
https://github.com/jina-ai/jina/issues/4476

### CODE REVIEW

File: docs/fundamentals/flow/client.md [hunk 1]
```diff
@@ -54,6 +54,52 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
 
 client.post('/endpoint')  # Empty
 ```
+### Specifying Host scheme
+
+To connect to the right `Flow` the client must now
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:59]: there is some verb missing here?

File: docs/fundamentals/flow/client.md [hunk 2]
```diff
@@ -54,6 +54,52 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
 
 client.post('/endpoint')  # Empty
 ```
+### Specifying Host scheme
+
+To connect to the right `Flow` the client must now 
+* the `protocol`he need to use to communicate with the `Flow
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:60]: I think the client is 'it' not he

File: docs/fundamentals/flow/client.md [hunk 3]
```diff
@@ -54,6 +54,52 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
...
+* the `host` and the `port`on which the Flow is exposed
+* if he needs to use `tls` encryption
+
+You can precise these parameters by passing a valid scheme to the `host` argument:
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:64]:
```suggestion
You can define these parameters by passing a valid URI scheme as part of the `host` argument:
```

File: docs/fundamentals/flow/client.md [hunk 4]
```diff
@@ -54,6 +54,52 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
...
+
+```python
+from jina import Client
+
+Client(host='https://my.awesome.flow:1234')
+Client(host='wss://my.awesome.flow:1234')
+Client(host='grpcs://my.awesome.flow:1234')
+```
+
+You can as well pass each relevant parameters as a keyword argument:
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:74]:
```suggestion
You can as well pass each relevant parameter as a keyword argument:
```

File: docs/fundamentals/flow/client.md [hunk 5]
```diff
@@ -54,6 +54,52 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
...
+You can use a mixe of both as well:
+
+```python
+Client(host='https://my.awesome.flow', port=1234)
+Client(host='my.awesome.flow:1234', protocol='http', tls=True)
+```
+
+````{admonition} Caution
+:class: caution
+You can't precise these parameters by keyword arugment and by host schem (You can't have two sources of truth)
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:92]:
```suggestion
You can't define these parameters by keyword arugment and by host scheme (You can't have two sources of truth)
```

**Review by jacobowitz — CHANGES_REQUESTED** [review]
Just some small language things in the docs, @JohannesMessner the master of documentation should probably also double check that part ;)

File: docs/fundamentals/flow/client.md [hunk 6]
```diff
@@ -54,6 +54,51 @@ client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
 
 client.post('/endpoint')  # Empty
 ```
+### Specifying Host scheme
+
+To connect to the right `Flow` the client must now the followings parameters:
```
Comment (jacobowitz) [inline docs/fundamentals/flow/client.md:59]:
```suggestion
To connect to the `Flow` the client has to specify the followings parameters:
```

There is still a verb missing here

**Review by jacobowitz — CHANGES_REQUESTED** [review]

File: docs/fundamentals/flow/client.md (hunk 6 above)
Comment (samsja) [reply docs/fundamentals/flow/client.md:59]: my bad again

**Review by jacobowitz — APPROVED** [review]
nice :+1:

**Pull request merged**